src/ReportsSync.php

```diff
@@ -208,669 +51,116 @@ public static function get_import_totals( $days, $skip_existing ) {
-			$chunk_size = (int) ceil( $range_size / $batch_size );
-
-			for ( $i = 0; $i < $batch_size; $i++ ) {
-				$batch_start = (int) ( $range_start + ( $i * $chunk_size ) );
-				$batch_end   = (int) min( $range_end, $range_start + ( $chunk_size * ( $i + 1 ) ) - 1 );
-
-				if ( $batch_start > $range_end ) {
-					return;
-				}
-
-				self::queue()->schedule_single(
-					$action_timestamp,
-					self::QUEUE_BATCH_ACTION,
-					array( $batch_start, $batch_end, $single_batch_action, $action_args ),
-					self::QUEUE_GROUP
-				);
-			}
-		} else {
-			// Otherwise, queue the single batches.
-			for ( $i = $range_start; $i <= $range_end; $i++ ) {
-				$batch_action_args = array_merge( array( $i ), $action_args );
-				self::queue()->schedule_single( $action_timestamp, $single_batch_action, $batch_action_args, self::QUEUE_GROUP );
-			}
-		}
-	}
-
-	/**
-	 * Queue an action to run after another.
-	 *
-	 * @param string $action Action to run after prerequisite.
-	 * @param array  $action_args Action arguments.
-	 * @param string $prerequisite_action Prerequisite action.
+	 * @return string
 	 */
-	public static function queue_dependent_action( $action, $action_args, $prerequisite_action ) {
-		$blocking_jobs = self::queue()->search(
-			array(
-				'status'   => 'pending',
-				'orderby'  => 'date',
-				'order'    => 'DESC',
-				'per_page' => 1,
-				'claimed'  => false,
-				'search'   => $prerequisite_action, // search is used instead of hook to find queued batch creation.
-				'group'    => self::QUEUE_GROUP,
-			)
-		);
-
-		$next_job_schedule = null;
-		$blocking_job_hook = null;
-
-		if ( $blocking_jobs ) {
-			$blocking_job      = current( $blocking_jobs );
-			$blocking_job_hook = $blocking_job->get_hook();
-			$next_job_schedule = $blocking_job->get_schedule()->next();
+	public static function regenerate_report_data( $days, $skip_existing ) {
+		if ( self::is_importing() ) {
+			return new \WP_Error( 'wc_admin_import_in_progress', __( 'An import is already in progress.  Please allow the previous import to complete before beginning a new one.', 'woocommerce-admin' ) );
 		}
 
-		// Eliminate the false positive scenario where the blocking job is
-		// actually another queued dependent action awaiting the same prerequisite.
-		// Also, ensure that the next schedule is a DateTime (it can be null).
-		if (
-			is_a( $next_job_schedule, 'DateTime' ) &&
-			( self::QUEUE_DEPENDENT_ACTION !== $blocking_job_hook )
-		) {
-			self::queue()->schedule_single(
-				$next_job_schedule->getTimestamp() + 5,
-				self::QUEUE_DEPENDENT_ACTION,
-				array( $action, $action_args, $prerequisite_action ),
-				self::QUEUE_GROUP
-			);
-		} else {
-			self::queue()->schedule_single( time() + 5, $action, $action_args, self::QUEUE_GROUP );
+		self::reset_import_stats( $days, $skip_existing );
+		foreach ( self::get_schedulers() as $scheduler ) {
+			$scheduler::schedule_action( 'import_batch_init', array( $days, $skip_existing ) );
```

Comment:
Same comment about checking for callable methods.

Reply:
Resolved via https://github.com/woocommerce/woocommerce-admin/pull/3285/commits/e7fd4e813f7c3618fd07d33243930e801c008a9e